tests/sentry/tasks/deletion/test_hybrid_cloud.py

```diff
@@ -274,3 +284,175 @@ def test_set_null_deletion_behavior(task_runner):
+    with assume_test_silo_mode_of(DiscoverSavedQuery, Monitor):
+        saved_query = DiscoverSavedQuery.objects.create(
+            name="disco-query",
+            organization=organization,
+            created_by_id=user.id,
+        )
+        monitor = Monitor.objects.create(
+            organization_id=organization.id,
+            project_id=project.id,
+            slug="test-monitor",
+            name="Test Monitor",
+            owner_user_id=user.id,
+        )
+
+    return dict(
+        user=user,
+        organization=organization,
+        project=project,
+        monitor=monitor,
+        group=group,
+        saved_query=saved_query,
+    )
+
+
+@django_db_all
+@region_silo_test
+def test_get_ids_for_tombstone_cascade_cross_db(task_runner):
+    data = setup_cross_db_deletion_data()
+
+    unaffected_data = []
+    for i in range(3):
+        unaffected_data.append(setup_cross_db_deletion_data())
+
+    user = data["user"]
+    user_id = user.id
+    monitor = data["monitor"]
+    with assume_test_silo_mode_of(User), outbox_runner():
+        user.delete()
+
+    tombstone = RegionTombstone.objects.get(
+        object_identifier=user_id, table_name=User._meta.db_table
+    )
+
+    highest_tombstone_id = RegionTombstone.objects.aggregate(Max("id"))
+    monitor_owner_field = Monitor._meta.get_field("owner_user_id")
+
+    with override_options({"hybrid_cloud.allow_cross_db_tombstones": True}):
+        ids, oldest_obj = get_ids_for_tombstone_cascade_cross_db(
+            tombstone_cls=RegionTombstone,
+            model=Monitor,
+            field=monitor_owner_field,
+            watermark_batch=WatermarkBatch(
+                low=0,
+                up=highest_tombstone_id["id__max"] + 1,
+                has_more=False,
+                transaction_id="foobar",
+            ),
+        )
+        assert ids == [monitor.id]
+        assert oldest_obj == tombstone.created_at
+
+
+@django_db_all
+@region_silo_test
+def test_get_ids_for_tombstone_cascade_cross_db_watermark_bounds(task_runner):
+    cascade_data = []
+    for i in range(4):
+        cascade_data.append(setup_cross_db_deletion_data())
+
+    unaffected_data = []
+    for i in range(3):
+        unaffected_data.append(setup_cross_db_deletion_data())
+
+    in_order_tombstones = []
+    for data in cascade_data:
+        user = data["user"]
+        user_id = user.id
+        with assume_test_silo_mode_of(User), outbox_runner():
+            user.delete()
+
+        in_order_tombstones.append(
+            RegionTombstone.objects.get(object_identifier=user_id, table_name=User._meta.db_table)
+        )
+
+    bounds_with_expected_results = [
+        (
+            {"low": 0, "up": in_order_tombstones[1].id},
+            [cascade_data[0]["monitor"].id, cascade_data[1]["monitor"].id],
+        ),
+        (
+            {"low": in_order_tombstones[1].id, "up": in_order_tombstones[2].id},
+            [cascade_data[2]["monitor"].id],
```

Comment:
If `cascade_data` has four sets of data in it, and all four users were deleted why are only three monitors being deleted?

Reply:
Oh that's a good point, let me fix this to span 3 instead. I'll also add a case that tests that the full set is deleted if the watermark low is set to -1